pkg/kubemark/controller.go

```diff
@@ -356,7 +356,11 @@ func (kubemarkCluster *kubemarkCluster) getHollowNodeName() (string, error) {
 		return "", err
 	}
 	for _, node := range nodes {
-		return node.Name, nil
+		for key := range node.ObjectMeta.Labels {
```

Comment:
You can directly list using a label-selector in the List() call above instead of filtering the results yourself.

Reply:
Done